src/components/Tabset/__test__/tabset.spec.js

```diff
@@ -43,4 +80,200 @@ describe('<Tabset />', () => {
+                <Tab label="Tab-3" name="tab-3" registerTab={registerTabMockFn} />
+            </Tabset>,
+        );
+        component.instance().scrollToSelectedTab = jest.fn();
+        const item1 = component.find('Tab[name="tab-1"]').find('a');
+        item1.simulate('click');
+        expect(component.instance().scrollToSelectedTab).toHaveBeenCalledWith('tab-1');
+    });
+    it('should not call updateButtonsVisibility function if any child is changed', () => {
+        isNotSameChildren.mockReset();
+        isNotSameChildren.mockReturnValue(false);
+        const component = mount(
+            <Tabset>
+                <Tab label="Tab-1" name="tab-1" registerTab={registerTabMockFn} />
+                <Tab label="Tab-2" name="tab-2" registerTab={registerTabMockFn} />
+                <Tab label="Tab-3" name="tab-3" registerTab={registerTabMockFn} />
+            </Tabset>,
+        );
+        component.instance().updateButtonsVisibility = jest.fn();
+        component.setProps();
+        expect(component.instance().updateButtonsVisibility).toHaveBeenCalledTimes(0);
+    });
+    it('should call updateButtonsVisibility function when a child is changed', () => {
+        isNotSameChildren.mockReset();
+        isNotSameChildren.mockReturnValue(true);
+        const component = mount(
+            <Tabset>
+                <Tab label="Tab-1" name="tab-1" registerTab={registerTabMockFn} />
+                <Tab label="Tab-2" name="tab-2" registerTab={registerTabMockFn} />
+                <Tab label="Tab-3" name="tab-3" registerTab={registerTabMockFn} />
+            </Tabset>,
+        );
+        component.instance().updateButtonsVisibility = jest.fn();
+        component.instance().isFirstTime = false;
+        component.setProps();
+        expect(component.instance().updateButtonsVisibility).toHaveBeenCalledTimes(1);
+    });
+    it('should call updateButtonsVisibility function and set isFirstTime to false when all children are registered and is first time', () => {
+        isNotSameChildren.mockReset();
+        isNotSameChildren.mockReturnValue(false);
+        const component = mount(
+            <Tabset activeTabName="tab-1">
+                <Tab label="Tab-1" name="tab-1" registerTab={registerTabMockFn} />
+                <Tab label="Tab-2" name="tab-2" registerTab={registerTabMockFn} />
+                <Tab label="Tab-3" name="tab-3" registerTab={registerTabMockFn} />
+            </Tabset>,
+        );
+        component.instance().updateButtonsVisibility = jest.fn();
+        component.instance().isFirstTime = true;
+        component.setState({
+            tabsetChildren: [
+                <Tab label="Tab-1" name="tab-1" registerTab={registerTabMockFn} />,
+                <Tab label="Tab-2" name="tab-2" registerTab={registerTabMockFn} />,
+                <Tab label="Tab-3" name="tab-3" registerTab={registerTabMockFn} />,
+            ],
+        });
+        component.setProps();
```

Comment:
review:
1. I think you don't need to set ```component.instance().isFirstTime = true;```
since always the component instance initialize this value to true. Then instead of initialize it manually to true test that initially it is true and then that it change to false after call setState.
2 . Maybe ```component.setProps();``` is not necessary since ```component.setState()``` already update the component.